### ISSUE TITLE
Feature Request: Use `the_excerpt` as is when `[ap_excerpt]` is called without length argument

### ISSUE THREAD

**Issue by mdingemanse** [Automattic/wordpress-activitypub#595]
### What

Ideally `[ap_excerpt]` would mirror WP excerpt logic by using the default cutoff for posts that do not have an excerpt set, but outputting the whole `the_excerpt` if it exists for a post.

If a user does set an explicit length argument it arguably should still be applied.

### Why

This would align with how WP themes generally deal with excerpts and excerpting. It is also a sensible default: trust the user who sets an except and align with their expectation that without any length specified, this will show up as is through AP.

(Current behaviour, unexpected for me: the excerpt is used but cutoff at default length. I need to set length explicitly in [ap_excerpt] shortcode call. But I don't know how long my excerpts will be so I end up setting some random high number. Not having to set anything would be a more sensible default.)

### How

Basically, for `[ap_excerpt]` without a length argument specified, check whether a post has an excerpt and if so, display that as is. If not, take first lines of `the_content` and truncate at default length.

If length _is_ explicitly specified in `[ap_excerpt length=340]` by the user then apply that to `the_content` or to`the_excerpt`, if that exists.

**Comment by mdingemanse** [comment]
No the issue should not be closed, it provides a reasoned proposal that brings the plugin closer to expected behaviour for WP users.
